### ISSUE TITLE
HDR videos are pixelated in Stremio internal video player in Linux

### ISSUE THREAD

**Issue by shscs911** [Stremio/stremio-bugs#549]
**Describe the bug**
- All HDR videos are pixelated in Stremio internal video player in Linux.
- Same videos are playing normally in mpv player without any issues.
- Issue is reproducible with `Hardware-accelerated decoding` option enabled as well as disabled.
- This issue started after the Stremio rebrand update. It was working normally before.

**To Reproduce**
1. Open Stremio Desktop.
2. Select any movie title and select 4K HDR quality.
3. Video plays with pixelation and skipped frames.

**Expected behavior**
1. Video should play normally without any issues, as Stremio uses mpv as backend in Linux.

**Screencapture**
* **You may need to download the screencaps to play them, as some browsers are not able to play them directly.**

* 4K HDR video in Stremio internal player

https://github.com/Stremio/stremio-bugs/assets/44034402/a9eb43ef-521c-4a59-8398-383b101d241f

![HDR Stremio](https://github.com/Stremio/stremio-bugs/assets/44034402/a0088b51-8762-441c-9d03-c17126251918)

* 4K HDR video in mpv

https://github.com/Stremio/stremio-bugs/assets/44034402/c7387f91-bd42-416d-8bad-497714d56c35

![HDR mpv](https://github.com/Stremio/stremio-bugs/assets/44034402/cca0b2a1-32a5-4633-b961-7f806dddefe5)

**Desktop (please complete the following information):**
 - OS: Arch Linux KDE
 - Browser: Mozilla Firefox 118.0b9 
 - Version:
   - Stremio: v4.4.160
   - Web UI: v4.4.164
   - Server: 4.19.0
   - mpv:
     - mpv v0.36.0-310-ged5c744aa9 Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
     - libplacebo version: v6.292.0-381-gc4c68fe6
     - FFmpeg version: n6.0
     - FFmpeg library versions:
     - libavutil       58.2.100
     - libavcodec      60.3.100
     - libavformat     60.3.100
     - libswscale      7.1.100
     - libavfilter     9.3.100
     - libswresample   4.10.100

**Comment by matijakovacevic** [comment]
Same here.

Although I must point out, it only happens on my Intel i7 11th gen laptop with Mesa Intel Xe Graphics.
On the AMD Ryzen one, all is working fine.

Operating System: KDE neon 5.27
KDE Plasma Version: 5.27.8
KDE Frameworks Version: 5.110.0

**Comment by shscs911** [comment]
> Same here.
> 
> Although I must point out, it only happens on my Intel i7 11th gen laptop with Mesa Intel Xe Graphics. On the AMD Ryzen one, all is working fine.
> 
> Operating System: KDE neon 5.27 KDE Plasma Version: 5.27.8 KDE Frameworks Version: 5.110.0

@matijakovacevic Was it working for you earlier? It was working for me previously with the mpv backend tonemapping the HDR content in Stremio to SDR.

**Comment by matijakovacevic** [comment]
> @matijakovacevic Was it working for you earlier? It was working for me previously with the mpv backend tonemapping the HDR content in Stremio to SDR.

@shscs911 yes, working before this brand

**Comment by shscs911** [comment]
> > @matijakovacevic Was it working for you earlier? It was working for me previously with the mpv backend tonemapping the HDR content in Stremio to SDR.
> 
> @shscs911 yep, working fine, before this re-brand update.
> 
> Although, not certain about this, but I think earlier version was installed as .deb package, whilst the rebrand I did over flatpak and removed the DEB package

For me the rebrand update was rolled out server-side and I didn't had to reinstall it. I noticed the issue immediately after  this rollout.

Though I cleaned up the leftover files and reinstalled it later as per the Stremio [guidelines](https://stremio.zendesk.com/hc/en-us/articles/360019466760-How-to-remove-Stremio-manually-on-Linux), the issue persists.

**Comment by TRtomasz** [comment]
stremio uses MPV that is already installed in your system, so if you had it installed before and update perhaps something changed in the mpv that caused this to break.

**Comment by shscs911** [comment]
You maybe right. I hadn't checked 4K HDR videos for a long time, but the HDR->SDR tonemapping works perfectly now in Stremio. I'm not sure if this is due to updating `mpv` or `Stremio`, because both of them have been updated after I've reported this issue.

@matijakovacevic Can you check if it's working for you now? My current applicaton version details are as follows:
 - Version:
   - Stremio: v4.4.164
   - Web UI: v4.4.167
   - Server: 4.20.2
   - mpv:
     - mpv v0.36.0-691-ge76660cc54 Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
     - libplacebo version: v6.338.0-43-ga59333c6
     - FFmpeg version: n6.0
     - FFmpeg library versions:
     - libavutil       58.2.100
     - libavcodec      60.3.100
     - libavformat     60.3.100
     - libswscale      7.1.100
     - libavfilter     9.3.100
     - libswresample   4.10.100

Closing the ticket as the issue's resolved now.

**Comment by sleeyax** [comment]
@shscs911 Did you adjust your mpv config for the HDR->SDR tonemapping or did it just work out of the box on the latest versions? I have the same issue as you described and pinned it down to an mpv issue on my system (Arch Linux), but I'm getting lost in this rabbit hole while this used to work fine in the past without any custom configuration.

**Comment by shscs911** [comment]
It works out of the box for me with the same configuration I've been using for a long time. I'm using [mpv-git](https://aur.archlinux.org/packages/mpv-git) package from AUR with minor changes from the vanilla config.

Following are the only video-related additions in the config:
```
vo=gpu-next
gpu-context=x11vk
hwdec=vaapi
video-sync=display-resample
```

**Comment by sleeyax** [comment]
No luck for me, I installed `mpv-git` (I used to have `mpv` installed) with a similar config and the same issue persists on Linux `6.1.61-1-lts` kernel.

It's not exactly pixelerated but there's this dusty 'noise' all over it which just looks terrible:

<img width="688" alt="image" src="https://github.com/Stremio/stremio-bugs/assets/30344294/394d1f8a-8b8a-4f14-b9b2-58a99a002e38">

It appears MPV is also having trouble with regular, non-HDR videos. Probably unrelated to this issue then, but annoying nonetheless. Thanks for your input so far, I appreciate it!
